test/jdk/tools/jpackage/helpers-test/jdk/jpackage/test/AnnotationsTest.java

```diff
@@ -300,26 +303,23 @@ public static Collection<Object[]> dateSupplier() {
         });
     }
 
-    private static void runTests(List<Class<? extends TestExecutionRecorder>> tests) {
+    private static void runTest(Class<? extends TestExecutionRecorder> test, Path workDir) {
         ACTUAL_TEST_DESCS.get().clear();
 
-        var expectedTestDescs = tests.stream()
-                .map(AnnotationsTest::getExpectedTestDescs)
-                .flatMap(x -> x)
+        var expectedTestDescs = getExpectedTestDescs(test)
                 // Collect in the map to check for collisions for free
                 .collect(toMap(x -> x, x -> ""))
                 .keySet();
 
-        var args = tests.stream().map(test -> {
-            return String.format("--jpt-run=%s", test.getName());
-        }).toArray(String[]::new);
+        var args = new String[] { String.format("--jpt-run=%s", test.getName()) };
 
         try {
-            Main.main(args);
+            Main.main(TestBuilder.build().workDirRoot(workDir), args);
             assertRecordedTestDescs(expectedTestDescs);
         } catch (Throwable t) {
             t.printStackTrace(System.err);
             System.exit(1);
+            return;
```

Comment:
Do we need `return` here?

Reply:
No. It sneaked in by an accident. Will fix it